calicoctl/commands/version.go

```diff
@@ -211,6 +211,10 @@ Description:
 
 	ci, err := client.ClusterInformation().Get(ctx, "default", options.GetOptions{})
 	if err != nil {
+		if strings.Contains(err.Error(), "does not exist") {
```

Comment:
Think we return a dedicated, reliable error type here; it should be `ErrorResourceDoesNotExist` from our errors package

Reply:
Is there an explicit contract somewhere for this, though?  If it's not robust, it doesn't feel obviously better than checking for a string.

Reply:
Seems we are checking NotExist error like this https://github.com/projectcalico/node/blob/master/pkg/lifecycle/startup/startup.go#L433

Reply:
Suggest printing out information before return nil so that user knows cluster information does not exist.

Reply:
I guess I can do that.  My concern is: there are at least 7 levels of `interface` usage between `client.ClusterInformation().Get` and what actually happens, and **none** of those interfaces says "if the resource is not found, the error will be `ErrorResourceDoesNotExist`.  So assuming that feels no less fragile to me than a hacky string check, and I do at least know that the string check will fix our CI because I've seen the string "does not exist" in the CI output.

(Aside: this feels like a structural weakness in golang.)